### PR TITLE
ENH/TST: ccf to optionally return confidence intervals

### PR DESCRIPTION
**Pull request by aglebov**
- Ability for `ccf` to return confidence intervals, similar to `acf` and `pacf`. The intervals are only returned if the function is called with a newly introduced argument `alpha`.
- Ability to limit the number of lags for which cross-correlations are returned by passing `nlags` to `ccf`, similar to how it is done in `acf` and `pacf`.
- Tests for `ccf`.
- Minor improvements to the documentation for `ccovf` and `ccf`

### CODE REVIEW

File: statsmodels/tsa/stattools.py [hunk 1]
```diff
@@ -1096,27 +1098,48 @@ def ccf(x, y, adjusted=True, fft=True):
...
     Returns
     -------
     ndarray
-        The cross-correlation function of x and y.
+        The cross-correlation function of x and y: the element at index k
+        is the correlation between {x[k], x[k+1], ..., x[n]} and {y[0], y[1], ..., y[m-k]},
+        where n and m are the lengths of x and y, respectively.
+    confint : ndarray, optional
+        Confidence intervals for the CCF at lags 0, 1, ..., nlags-1. Shape
+        (nlags, 2). Returned if alpha is not None.
```
Comment (josef-pkt) [inline statsmodels/tsa/stattools.py:1122]: This should mention that it uses sqrt(nobs) as standard error for confidence interval

**Comment by josef-pkt** [comment]
looks good to me

Is there an "official" reference for the confidence interval standard error?

I found https://ideas.repec.org/c/boc/bocode/s355401.html in a quick search which uses 2 * sqrt(nobs) (i.e. fixed alpha but same stderror)

can you squash the commits?

File: statsmodels/tsa/stattools.py (hunk 1 above)
Comment (aglebov) [reply statsmodels/tsa/stattools.py]: Done

**Comment by aglebov** [comment]
> Is there an "official" reference for the confidence interval standard error?

I used the reference quoted in the docstring for `plot_acf`: Brockwell and Davis' "Introduction to Time Series and Forecasting". In the third edition of the book (2016), the formula is on page 242:

bartlett

Then you get $\lim_{n \to \infty} n \mathrm{Var}(\hat\rho_{12}(h)) = 1$ by taking $h=k$ and assuming all rhos are zero except for $\rho_{11}(0)=\rho_{22}(0)=1$.

> I found https://ideas.repec.org/c/boc/bocode/s355401.html in a quick search which uses 2 * sqrt(nobs) (i.e. fixed alpha but same stderror)

I was coming from the implementation in R: https://github.com/SurajGupta/r-source/blob/master/src/library/stats/R/acf.R#L135

> can you squash the commits?

Done

**Comment by josef-pkt** [comment]
Very good, merging

Thanks for the PR

**Pull request merged**